cpp/tests/strings/factories_test.cu

```diff
@@ -176,3 +178,26 @@ TEST_F(StringsFactoriesTest, EmptyStringsColumn)
     results = cudf::make_strings_column( d_strings );
     cudf::test::expect_strings_empty(results->view());
 }
+
+TEST_F(StringsFactoriesTest, CreateOffsets)
+{
+  std::vector<std::string> strings = {"this", "is", "a", "column", "of", "strings"};
+  cudf::test::strings_column_wrapper sw = { strings.begin(), strings.end() };                                                
+  cudf::column_view col(sw);
+  std::vector<cudf::size_type> indices{ 2,6 };
+  auto result = cudf::experimental::slice(col, indices);
+  auto expected_index = indices[0];
+  for(size_t idx=0; idx<result.size(); idx++)
+  {     
+    auto strings_data = cudf::strings::create_offsets(cudf::strings_column_view(result[idx]));
+    thrust::host_vector<char> h_chars(strings_data.first);
+    thrust::host_vector<cudf::size_type> h_offsets(strings_data.second);
+    for( size_t jdx=0; jdx < h_offsets.size()-1; ++jdx )
+    {
+        auto offset = h_offsets[jdx];
+        auto length = h_offsets[jdx+1] - offset;
+        std::string str(h_chars.data()+offset, length);
+        EXPECT_EQ(str,strings[expected_index++]);
```

Comment:
I would find this test easier to follow if you created an expected_strings array that contained the expected result and compared them.  It took some digging to ascertain that this test does what is intended.  Since it's a unit test, I think something like:
```
std::vector<std::string> expected_strings = { "is", "a", "column", "of" };
 ...
EXPECT_EQ(str, expected_strings[jdx]);
```
Might be a little easier to interpret.  It would also allow you to test multiple slices if you wanted.